src/app.ts

```diff
@@ -59,6 +60,23 @@ async function routes(fastify: FastifyInstance) {
   }
 }
 
+export const validatorCompiler: FastifySchemaCompiler<ZodAny> =
+  ({ schema }) =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (data): any => {
+    try {
+      return { value: schema.parse(data) };
+    } catch (error) {
+      if (error instanceof ZodError && error.errors.length) {
+        const firstError = error.errors[0];
+        return {
+          error: new Error(`Invalid ${firstError.path.join('.')}: ${error.errors[0].message}`),
```

Comment:
prefer  throw `throw fastify.httpErrors.badRequest(error.message);` instead of default Error

Reply:
Updated at 56a7efd